angular-app/src/app/project-detail/project-detail.component.ts

```diff
@@ -57,4 +58,24 @@ export class ProjectDetailComponent implements OnInit, OnChanges {
     }
     this.setBlobstoreUploadUrl();
   }
+
+  deleteFile(project: Project, file: ProjectFile) {
+    this.projectService.deleteFile(project, file);
+  }
+
+  deleteProject(project: Project) {
+    const dialogRef = this.dialog.open(ConfirmationDialogComponent, {
```

Comment:
So, the default for your button labels in the confirmation dialog is "Continue" and "Cancel", which is fine as far as defaults go. But it might be nice to be able to customize the labels depending on the context of the confirmation. It's explicitly clear to a user if the button is labeled "Delete" that that will cause their project to be deleted. And for an operation like delete, which presumably can't be undone, the clearer the prompt, the better.

Reply:
Added. We might also consider further optional customization to indicate which option should receive initial focus/coloring